elastic/kibana
# [Security Solution][Expandable flyout] - add onClose callback logic to Security Solution application

## Summary

This PR adds the propagation of the expandable flyout `onClose` to all components in Security Solution. The solution chose leverages window events.

We have 2 events attached to the window:
- one for the normal (non-timeline) flyout, called `expandable-flyout-on-close-SecuritySolution`
- one for the timeline flyout, called `expandable-flyout-on-close-Timeline`

When a flyout is closed, the `kbn-expandable-flyout` packages propagates an `onClose` event, which then in Security Solution dispatches a custom event (using `window.dispatchEvent`) with the key of which flyout is being closed.

A hook is available for developers to catch the `onClose` event, without having to know which flyout (normal or timeline) is being closed. Developers just have to use the hook by passing a callback to it, which will be run when the event is dispatched. At the time, the event will also be removed from the window.

### How to test

An implementation was done in this PR to show how it works and verify it works correctly:
- make sure that `unifiedComponentsInTimelineEnabled` is in you `kibana.yml` file
- open a timeline
- click on the expand icon to open the expandable flyout
- close the flyout

The row of the alert that was open should not have the highlight after closing the flyout. Similarly, the expand icon should change back.

https://github.com/user-attachments/assets/4e7e3a6f-995b-4af2-aa4e-0206c3e462f2

### Checklist


https://github.com/elastic/kibana/issues/179520
